tls/s2n_client_hello_request.c

```diff
@@ -41,6 +41,13 @@ S2N_RESULT s2n_client_hello_request_recv(struct s2n_connection *conn)
     RESULT_ENSURE_REF(conn->config);
     RESULT_GUARD(s2n_client_hello_request_validate(conn));
 
+    /* Maintain the old s2n-tls behavior by default.
+     * Traditionally, s2n-tls has just ignored all hello requests.
+     */
+    if (!conn->config->renegotiate_request_cb) {
```

Comment:
The RFC says the hello request MAY be ignored by the client. Is there a benefit to sending this alert at all, even if the callback rejects the renegotiation?

Reply:
It will let servers that require renegotiation gracefully close the connection without just timing out, which could make diagnosing failures due to renegotiation easier.

But I do also plan to add an option for applications implementing the callback to ignore the request: https://github.com/aws/s2n-tls/pull/3556 Applications can do that right now with S2N_RENEGOTIATE_ACCEPT, but that's very confusing naming.

Reply:
Ah okay, that makes sense. Thanks!